framework/libra-framework/sources/ol_sources/lockbox.move

```diff
@@ -0,0 +1,464 @@
+    let (_,_) = idx_by_duration(bob_addr, 1*12);
+  }
+
+  #[test(framework = @0x1, bob_sig = @0x10002)]
+  fun creates_lockbox(framework: &signer, bob_sig: &signer) acquires SlowWalletV2 {
+    let bob_addr = signer::address_of(bob_sig);
+
+    let coin = test_setup(framework, 23);
+
+    add_to_or_create_box(bob_sig, coin, 1*12);
+
+    // see if it exists
+    let (found, idx) = idx_by_duration(bob_addr, 1*12);
+    assert!(found, 7357002);
+    assert!(idx == 0, 7357003);
+
+    let balance_one = balance_duration(bob_addr, 1*12);
+    assert!(balance_one == 23, 7357004);
+    let balanace_all = balance_all(bob_addr);
+    assert!(balanace_all == 23, 7357005);
+  }
+
+  #[test(framework = @0x1, bob_sig = @0x10002)]
+  fun adds_to_lockbox(framework: &signer, bob_sig: &signer) acquires SlowWalletV2 {
+    let bob_addr = signer::address_of(bob_sig);
+    let coin = test_setup(framework, 123);
+    let split_coin = libra_coin::extract(&mut coin, 23);
+
+    add_to_or_create_box(bob_sig, split_coin, 1*12);
+
+    // see if it exists
+    let (found, idx) = idx_by_duration(bob_addr, 1*12);
+    assert!(found, 7357002);
+    assert!(idx == 0, 7357003);
+
+    let bal = balance_all(bob_addr);
+    assert!(bal == 23, 7357004);
+
+    // remainder of coin should be 100
+    add_to_or_create_box(bob_sig, coin, 1*12);
+
+    // see if it exists
+    let (found, idx) = idx_by_duration(bob_addr, 1*12);
+    assert!(found, 7357005);
+    assert!(idx == 0, 7357006);
+
+    let bal2 = balance_all(bob_addr);
+    assert!(bal2 == 123, 7357007);
+  }
+
+  #[test(framework = @0x1, bob_sig = @0x10002)]
+  #[expected_failure(abort_code = 65545, location = ol_framework::lockbox)]
+  fun test_non_standard_duration(framework: &signer, bob_sig: &signer) acquires SlowWalletV2 {
+    let bob_addr = signer::address_of(bob_sig);
+    let coin = test_setup(framework, 100);
+
+    // Try to create a lockbox with a non-standard duration (5*12 months)
+    // This should fail because 5*12 is not in DEFAULT_LOCKS
+    add_to_or_create_box(bob_sig, coin, 5*12);
+
+    // These assertions should never be reached because the above call should fail
+    let (found, _) = idx_by_duration(bob_addr, 5*12);
+    assert!(!found, 7357001);
+  }
+
+  #[test(framework = @0x1, bob_sig = @0x10002)]
+  fun test_standard_duration(framework: &signer, bob_sig: &signer) acquires SlowWalletV2 {
+    let bob_addr = signer::address_of(bob_sig);
+    let coin = test_setup(framework, 100);
+
+    // Try to create a lockbox with a standard duration (4*12 months)
+    // This should succeed because 4*12 is in DEFAULT_LOCKS
+    add_to_or_create_box(bob_sig, coin, 4*12);
+
+    // Verify the lockbox was created with the standard duration
+    let (found, idx) = idx_by_duration(bob_addr, 4*12);
+    assert!(found, 7357001);
+    assert!(idx == 0, 7357002);
+
+    // Verify the balance
+    let balance = balance_duration(bob_addr, 4*12);
+    assert!(balance == 100, 7357003);
+  }
+}
```

Comment:
In general test coverage feels a bit low. E.g. should we test a locked account all the way to the end of its lifetime? Test moving unlocked funds from a lockbox account then continue to drip it after. Test moving locked funds to another lockbox account, then move again to a third. Try to extract more coins from a lockbox account than are currently unlocked.